### PR TITLE
`prevent-link-loss` - Restore support for all fields

### LINKED ISSUE #7076: `prevent-link-loss`: Broken on new field
### Description

Due to:

- #7011 

It works on the "New issue" forms (although it's not spaced correctly), but not when adding new comments or review comments.

### How to replicate the issue + URL

Here's the link to test with:

```
https://github.com/refined-github/refined-github/pull/6954/commits/32d1c8b2e1b6971709fe273cfdd1f959b51e8d85
```

### New issue forms ✅ (they use the old form)

https://github.com/refined-github/refined-github/issues/new?assignees=&labels=bug&projects=&template=1_bug_report.yml

Screenshot 7

### New comment

Here

Screenshot 8

### New review

https://github.com/refined-github/sandbox/pull/4/files#review-changes-modal

Screenshot 9

### New review comment

https://github.com/refined-github/sandbox/pull/4/files

Screenshot 10

### Extension version

23.11.3

### Browser(s) used

Safari

### PR DESCRIPTION
**Pull request by cheap-glitch**
Fixes #7076 

## Test URLs

Test link:
```
https://github.com/refined-github/refined-github/pull/6954/commits/32d1c8b2e1b6971709fe273cfdd1f959b51e8d85
```

- new issue form: https://github.com/refined-github/refined-github/issues/new?assignees=&labels=bug&projects=&template=1_bug_report.yml
- new comment form (here)
- review & review comment forms: https://github.com/refined-github/sandbox/pull/4/files#review-changes-modal

### New issue
image

### New comment
image

### New review
image

### New review comment
image

### CODE REVIEW

File: source/features/prevent-link-loss.tsx [hunk 1]
```diff
@@ -25,10 +25,10 @@ function handleButtonClick({currentTarget: fixButton}: React.MouseEvent<HTMLButt
 	replaceFieldText(field, prCommitUrlRegex, preventPrCommitLinkLoss);
 	replaceFieldText(field, prCompareUrlRegex, preventPrCompareLinkLoss);
 	replaceFieldText(field, discussionUrlRegex, preventDiscussionLinkLoss);
-	fixButton.parentElement!.remove();
+	fixButton.parentElement!.parentElement!.remove();
```
[the remark this answers is not in the document]
Comment (fregante) [reply source/features/prevent-link-loss.tsx:28]: Yes please: https://github.com/sindresorhus/eslint-plugin-unicorn/issues/345

File: source/features/prevent-link-loss.tsx [hunk 2]
```diff
@@ -53,28 +53,29 @@ function isVulnerableToLinkLoss(value: string): boolean {
...
-	} else if (pageDetect.isNewIssue() || pageDetect.isNewRelease() || pageDetect.isCompare()) {
-		$('file-attachment', field.form!)!.append(
-			getUI(field, 'mt-2', 'mx-0', 'mx-md-2'),
-		);
+function updateUI({delegateTarget: field}: DelegateEvent<Event, HTMLTextAreaElement>): void {
+	if (isVulnerableToLinkLoss(field.value)) {
+		$('file-attachment .js-write-bucket', field.form!)!.append(getUI(field));
 	} else {
-		$('.form-actions', field.form!)!.before(
-			getUI(field, 'mx-md-2', 'mb-2'),
```
Comment (fregante) [inline source/features/prevent-link-loss.tsx:65]: Has this difference been eliminated now? We still have two different forms, are they both covered by the new selector?

File: source/features/prevent-link-loss.tsx [hunk 3]
```diff
@@ -53,28 +53,29 @@ function isVulnerableToLinkLoss(value: string): boolean {
...
-		'form.js-new-comment-form textarea',
-		'textarea.comment-form-textarea',
-	].join(','), 'input', updateUI, {signal});
+	const textFieldsSelector = `form:is(
+		#new_issue,
+		#new_release,
+		#new_comment_form,
+		#pull_requests_submit_review,
+		.js-inline-comment-form
+	) textarea`;
```
Comment (fregante) [inline source/features/prevent-link-loss.tsx:75]:
```suggestion
	const textFieldsSelector = `
		form:is(
			#new_issue,
			#new_release,
			#new_comment_form,
			#pull_requests_submit_review,
			.js-inline-comment-form
		) textarea
	`;
```

**Review by fregante — APPROVED** [review]
Untested

File: source/features/prevent-link-loss.tsx [hunk 4]
```diff
@@ -53,28 +53,29 @@ function isVulnerableToLinkLoss(value: string): boolean {
...
+
+	delegate(textFieldsSelector, 'focusin', updateUI, {signal});
```
[the remark this answers is not in the document]
Comment (fregante) [reply source/features/prevent-link-loss.tsx:79]: I think here we're using the wrong selector, we have other features that target this field with less strict selectors. I think `text-expander textarea`, `.js-comment-field`, OR `[name="comment[body]"]` should work just as well.

File: source/features/prevent-link-loss.tsx (hunk 2 above)
Comment (cheap-glitch) [reply source/features/prevent-link-loss.tsx:65]: > Has this difference been eliminated now?

Looks like it, at least it works fine in all the forms I've tested with.

**Comment by cheap-glitch** [comment]
Tested again with the simplified selector in all the forms, works as intended.

**Pull request merged**